spec/lcm/integration/user_provisioning_extended_spec.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'fixtures/user_provisioning_fixtures'
+require_relative 'support/user_provisioning_helper'
+require_relative 'support/configuration_helper'
+require_relative 'support/s3_helper'
+require_relative '../../../lib/gooddata/models/user_filters/user_filter_builder'
+require_relative '../../../lib/gooddata/lcm/lcm2'
+
+describe 'the extended user provisioning functionality' do
+  before(:all) do
+    @fixtures = Fixtures::UserProvisioningFixtures.new projects_amount: 2,
+                                                       user_amount: 2
+    @project_fixtures = Fixtures::ProjectFixtures.new rest_client: @fixtures[:rest_client], domain: @fixtures[:domain]
+  end
+
+  after(:all) do
+    @fixtures.teardown
+    @project_fixtures.teardown
+  end
+
+  describe 'when provisioning in the declarative mode' do
+    it 'removes MUFs from projects not mentioned in the input' do
+      extra_client, extra_project = @project_fixtures[:clients].first
+      extra_mufs = Support::UserProvisioningHelper.random_muf_data client: extra_client, project: extra_project, users: @fixtures[:users]
+      builder_config = {
+        user_column: 'login',
+        labels: [{
+                   column: 'value',
+                   label: extra_mufs.first[:label_id]
+                 }]
+      }
+      extra_project.add_data_permissions GoodData::UserFilterBuilder.get_filters(GoodData::LCM2.convert_to_smart_hash(extra_mufs), GoodData::LCM2.convert_to_smart_hash(builder_config))
+
+      mufs = @project_fixtures[:clients].map do |client, project|
```

Comment:
warning: Similar code found in user_provisioning_end_to_end_spec.rb:46, user_provisioning_extended_spec.rb:33 (mass = 44) (flay)